tools/integration-test/src/tests/next/timeout_transfer.rs

```diff
@@ -0,0 +1,100 @@
+use ibc_relayer_framework::core::impls::packet_relayers::timeout_unordered_packet::BaseTimeoutUnorderedPacketRelayer;
+use ibc_relayer_framework::core::traits::packet_relayers::timeout_unordered_packet::TimeoutUnorderedPacketRelayer;
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::util::random::random_u64_range;
+
+use crate::tests::next::context::build_cosmos_relay_context;
+
+#[test]
+fn test_ibc_transfer_next() -> Result<(), Error> {
+    run_binary_channel_test(&IbcTransferTest)
+}
+
+pub struct IbcTransferTest;
+
+impl TestOverrides for IbcTransferTest {
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+}
+
+impl BinaryChannelTest for IbcTransferTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        _relayer: RelayerDriver,
+        chains: ConnectedChains<ChainA, ChainB>,
+        channel: ConnectedChannel<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        let relay_context = build_cosmos_relay_context(&chains);
+        let relayer = BaseTimeoutUnorderedPacketRelayer;
+
+        let runtime = chains.node_a.value().chain_driver.runtime.as_ref();
+
+        let denom_a = chains.node_a.denom();
+
+        let wallet_a = chains.node_a.wallets().user1().cloned();
+        let wallet_b = chains.node_b.wallets().user1().cloned();
+
+        let balance_a = chains
+            .node_a
+            .chain_driver()
+            .query_balance(&wallet_a.address(), &denom_a)?;
+
+        let a_to_b_amount = random_u64_range(1000, 5000);
+
+        info!(
+            "Sending IBC timeout from chain {} to chain {} with amount of {} {}",
+            chains.chain_id_a(),
+            chains.chain_id_b(),
+            a_to_b_amount,
+            denom_a
+        );
+
+        let chain_a_height = chains.handle_a.query_latest_height().unwrap();
```

Comment:
Should we actually be querying for chain b's height and passing that in as the `destination_height` parameter when calling `relay_timeout_unordered_packet`?

Reply:
We can only query for the latest height of a chain. But to build a proof at height H, we need the chain to reach at least height H+1. So either we have to wait for the chain to reach H+1, or we can build the proof now at height H-1, assuming that the proof is still valid at that height.